### ISSUE TITLE
[Issue]: RHEL 8 - Sample Failure - videoDecodeBatch

### ISSUE THREAD

**Issue by kiritigowda** [ROCm/rocDecode#341]
### Problem Description

```

======== CMake output     ======
The C compiler identification is GNU 8.5.0
The CXX compiler identification is GNU 8.5.0
Detecting C compiler ABI info
Detecting C compiler ABI info - done
Check for working C compiler: /usr/bin/cc - skipped
Detecting C compile features
Detecting C compile features - done
Detecting CXX compiler ABI info
Detecting CXX compiler ABI info - done
Check for working CXX compiler: /usr/bin/c++ - skipped
Detecting CXX compile features
Detecting CXX compile features - done
Found PkgConfig: /usr/bin/pkg-config (found version "1.4.2") 
Checking for module 'libavcodec'
  Found libavcodec, version 58.134.100
Checking for module 'libavformat'
  Found libavformat, version 58.76.100
Checking for module 'libavutil'
  Found libavutil, version 56.70.100
-- Using FFMPEG -- 
	Libraries:/usr/local/lib/libavcodec.so;/usr/local/lib/libavformat.so;/usr/local/lib/libavutil.so 
	Includes:/usr/local/include
-- videodecodebatch: Using rocDecode -- 
	Libraries:/opt/rocm/lib/librocdecode.so 
	Includes:/opt/rocm/include/rocdecode
Check if compiler accepts -pthread
Check if compiler accepts -pthread - yes
Found Threads: TRUE  
Configuring done
Generating done
Build files have been written to: /var/jenkins_home/workspace/main_testonly_rocDecode_develop/NKH6Ljlp5/rocdecode/build/release/rocdecode-test/videoDecodeBatch
======== End CMake output ======
Change Dir: /var/jenkins_home/workspace/main_testonly_rocDecode_develop/NKH6Ljlp5/rocdecode/build/release/rocdecode-test/videoDecodeBatch

Run Clean Command:/usr/bin/gmake -f Makefile clean

Run Build Command(s):/usr/bin/gmake -f Makefile && [ 33%] Building CXX object CMakeFiles/videodecodebatch.dir/videodecodebatch.cpp.o
[ 66%] Building CXX object CMakeFiles/videodecodebatch.dir/opt/rocm/share/rocdecode/utils/rocvideodecode/roc_video_dec.cpp.o
[100%] Linking CXX executable videodecodebatch
ld.lld: error: undefined symbol: std::filesystem::status(std::filesystem::__cxx11::path const&)
>>> referenced by videodecodebatch.cpp
>>>               CMakeFiles/videodecodebatch.dir/videodecodebatch.cpp.o:(ParseCommandLine(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char>>&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char>>&, int&, int&, bool&, OutputSurfaceMemoryType_enum&, int, char**))

ld.lld: error: undefined symbol: std::filesystem::remove_all(std::filesystem::__cxx11::path const&)
>>> referenced by videodecodebatch.cpp
>>>               CMakeFiles/videodecodebatch.dir/videodecodebatch.cpp.o:(ParseCommandLine(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char>>&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char>>&, int&, int&, bool&, OutputSurfaceMemoryType_enum&, int, char**))

ld.lld: error: undefined symbol: std::filesystem::create_directory(std::filesystem::__cxx11::path const&)
>>> referenced by videodecodebatch.cpp
>>>               CMakeFiles/videodecodebatch.dir/videodecodebatch.cpp.o:(ParseCommandLine(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char>>&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char>>&, int&, int&, bool&, OutputSurfaceMemoryType_enum&, int, char**))

ld.lld: error: undefined symbol: std::filesystem::__cxx11::directory_iterator::directory_iterator(std::filesystem::__cxx11::path const&, std::filesystem::directory_options, std::error_code*)
>>> referenced by videodecodebatch.cpp
>>>               CMakeFiles/videodecodebatch.dir/videodecodebatch.cpp.o:(main)

ld.lld: error: undefined symbol: std::filesystem::__cxx11::directory_iterator::operator++()
>>> referenced by videodecodebatch.cpp
>>>               CMakeFiles/videodecodebatch.dir/videodecodebatch.cpp.o:(main)

ld.lld: error: undefined symbol: std::filesystem::__cxx11::directory_iterator::operator*() const
>>> referenced by videodecodebatch.cpp
>>>               CMakeFiles/videodecodebatch.dir/videodecodebatch.cpp.o:(main)

ld.lld: error: undefined symbol: std::filesystem::__cxx11::path::_M_split_cmpts()
>>> referenced by videodecodebatch.cpp
>>>               CMakeFiles/videodecodebatch.dir/videodecodebatch.cpp.o:(std::filesystem::__cxx11::path::path<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char>>, std::filesystem::__cxx11::path>(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char>> const&, std::filesystem::__cxx11::path::format))
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
gmake[2]: *** [CMakeFiles/videodecodebatch.dir/build.make:119: videodecodebatch] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:83: CMakeFiles/videodecodebatch.dir/all] Error 2
gmake: *** [Makefile:91: all] Error 2



83% tests passed, 1 tests failed out of 6

Total Test time (real) = 120.69 sec

The following tests FAILED:
	  6 - video_decodeBatch (Failed)

```

### Operating System

RHEL 8

### CPU

ALL

### GPU

AMD Instinct MI300X

### ROCm Version

ROCm 6.1.0

**Comment by kiritigowda** [comment]
Fix issued - #342

**Comment by kiritigowda** [comment]
```
    Start 6: video_decodeBatch

6: Test command: /opt/cmake-3.25.2/bin/ctest "--build-and-test" "/opt/rocm/share/rocdecode/samples/videoDecodeBatch" "/var/jenkins_home/workspace/main_testonly_rocDecode_develop/zsJVqXy2K/rocdecode/build/release/rocdecode-test/videoDecodeBatch" "--build-generator" "Unix Makefiles" "--test-command" "videodecodebatch" "-i" "/opt/rocm/share/rocdecode/video/" "-t" "2"
6: Working Directory: /var/jenkins_home/workspace/main_testonly_rocDecode_develop/zsJVqXy2K/rocdecode/build/release/rocdecode-test
6: Test timeout computed to be: 1500
6: Internal cmake changing into directory: /var/jenkins_home/workspace/main_testonly_rocDecode_develop/zsJVqXy2K/rocdecode/build/release/rocdecode-test/videoDecodeBatch
6: ======== CMake output     ======
6: The C compiler identification is GNU 8.5.0
6: The CXX compiler identification is GNU 8.5.0
6: Detecting C compiler ABI info
6: Detecting C compiler ABI info - done
6: Check for working C compiler: /usr/bin/cc - skipped
6: Detecting C compile features
6: Detecting C compile features - done
6: Detecting CXX compiler ABI info
6: Detecting CXX compiler ABI info - done
6: Check for working CXX compiler: /usr/bin/c++ - skipped
6: Detecting CXX compile features
6: Detecting CXX compile features - done
6: Found PkgConfig: /usr/bin/pkg-config (found version "1.4.2") 
6: Checking for module 'libavcodec'
6:   Found libavcodec, version 58.134.100
6: Checking for module 'libavformat'
6:   Found libavformat, version 58.76.100
6: Checking for module 'libavutil'
6:   Found libavutil, version 56.70.100
6: -- Using FFMPEG -- 
6: 	Libraries:/usr/local/lib/libavcodec.so;/usr/local/lib/libavformat.so;/usr/local/lib/libavutil.so 
6: 	Includes:/usr/local/include
6: -- videodecodebatch: Using rocDecode -- 
6: 	Libraries:/opt/rocm/lib/librocdecode.so 
6: 	Includes:/opt/rocm/include/rocdecode
6: Check if compiler accepts -pthread
6: Check if compiler accepts -pthread - yes
6: Found Threads: TRUE  
6: Configuring done
6: Generating done
6: Build files have been written to: /var/jenkins_home/workspace/main_testonly_rocDecode_develop/zsJVqXy2K/rocdecode/build/release/rocdecode-test/videoDecodeBatch
6: ======== End CMake output ======
6: Change Dir: /var/jenkins_home/workspace/main_testonly_rocDecode_develop/zsJVqXy2K/rocdecode/build/release/rocdecode-test/videoDecodeBatch
6: 
6: Run Clean Command:/usr/bin/gmake -f Makefile clean
6: 
6: Run Build Command(s):/usr/bin/gmake -f Makefile && [ 33%] Building CXX object CMakeFiles/videodecodebatch.dir/videodecodebatch.cpp.o
6: [ 66%] Building CXX object CMakeFiles/videodecodebatch.dir/opt/rocm/share/rocdecode/utils/rocvideodecode/roc_video_dec.cpp.o
6: [100%] Linking CXX executable videodecodebatch
6: [100%] Built target videodecodebatch
6: 
6: Running test command: "/var/jenkins_home/workspace/main_testonly_rocDecode_develop/zsJVqXy2K/rocdecode/build/release/rocdecode-test/videoDecodeBatch/videodecodebatch" "-i" "/opt/rocm/share/rocdecode/video/" "-t" "2"
6: info: Number of threads: 2
6: info: decoding AMD_driving_virtual_20-H264.mp4 using GPU device 0 - AMD Instinct MI250X[gfx90a:sramecc+:xnack-] on PCI bus c6:00.0
6: info: decoding AMD_driving_virtual_20-H265.mp4 using GPU device 0 - AMD Instinct MI250X[gfx90a:sramecc+:xnack-] on PCI bus c6:00.0
6: Input Video Information
6: 	Codec        : AVC/H.264
6: 	Frame rate   : 60000/2002 = 29.97 fps
6: 	Sequence     : Progressive
6: 	Coded size   : [2048, 1024]
6: 	Display area : [0, 0, 2048, 1024]
6: 	Chroma       : YUV 420
6: 	Bit depth    : 8
6: Video Decoding Params:
6: 	Num Surfaces : 7
6: 	Crop         : [0, 0, 2048, 1024]
6: 	Resize       : 2048x1024
6: 
6: Input Video Information
6: 	Codec        : H.265/HEVC
6: 	Frame rate   : 30000/1001 = 29.97 fps
6: 	Sequence     : Progressive
6: 	Coded size   : [2048, 1024]
6: 	Display area : [0, 0, 2048, 1024]
6: 	Chroma       : YUV 420
6: 	Bit depth    : 8
6: Video Decoding Params:
6: 	Num Surfaces : 7
6: 	Crop         : [0, 0, 2048, 1024]
6: 	Resize       : 2048x1024
6: 
6: info: Total frame decoded: 1202
6: info: avg decoding time per frame: 1.1104 ms
6: info: avg FPS: 900.576
6: 
6/6 Test #6: video_decodeBatch ................   Passed    2.93 sec

100% tests passed, 0 tests failed out of 6
```
